services/wireguard/resources/handler.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package resources
+
+import "net"
+
+// Handler is mock wireguard resource handler.
+// It will manage lists of network interfaces names, IP addresses and port for endpoints.
+type Handler struct{}
+
+// AllocateInterface provides available name for the wireguard network interface.
+func (h *Handler) AllocateInterface() string { return "myst0" }
+
+// AllocateIP provides available IP address for the wireguard connection.
+func (h *Handler) AllocateIP() net.IPNet {
+	ip, ipnet, _ := net.ParseCIDR("10.182.47.1/24")
```

Comment:
Since you will allocate the same address for Provider and for Consumer, network stack will complain about duplicate address and will refuse to work :) We could allocate a subnet, and use .1 for provider. 
Since we will have to allocate separate subnet for each new wg peer (1 device per peer), we can always set .1 for provider and .2 for consumer, just different subnet. This way we will not have to deal with isolation and IP address tracking within subnet, but will need to track used subnets. We could allocate /30 subnets, but this will complicate counting. Thus lets stick with allocation if 10.182.x.1 (2) addresses for now.

Reply:
Of course, it will not be the same IP addresses for consumer and provider. 
This PR is about a frame for wireguard service. There is no consumer. 

It's enough to have this mocked IP address for this PR. Let's add IP-addresses handling in the next PR, when it will be needed. 

The idea here is to have a pool of IP addresses which can be assigned to the peers(provider or consumer). 

We were planning to start with a single network interface for multiple session, so provider here getting 10.182.47.1, and other consumers will get other available IP-addresses.

Reply:
This is where you are wrong. We were not going to have multiple peers per single interface. Whole idea is to have 1 interface per peer. We should declare this as soon as possible so not to have further misconceptions.

Reply:
Ok, then I got you wrong.

Reply:
Some motivation for this is described here: https://github.com/mysteriumnetwork/node/wiki/Wireguard:-multiple-vs-single-peer-per-wg-interface

Reply:
Removed `IP` and now sending only subnets called AllowedIPs.